src/utils/time-code.js

```diff
@@ -3,16 +3,39 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 // @flow
+import { sendAnalytics } from './analytics';
 
+const MAX_TIMINGS_PER_LABEL = 3;
+const _timingsPerLabel = {};
+
+/**
+ * We care about timing information. This function helps log and collect information
+ * about how fast our functions are.
+ */
 export function timeCode<T>(label: string, codeAsACallback: () => T): T {
-  if (
-    typeof performance !== 'undefined' &&
-    process.env.NODE_ENV === 'development'
-  ) {
+  if (typeof performance !== 'undefined') {
     const start = performance.now();
     const result = codeAsACallback();
     const elapsed = Math.round(performance.now() - start);
-    console.log(`${label} took ${elapsed}ms to execute.`);
+
+    // Only log timing information in development mode.
+    if (process.env.NODE_ENV === 'development') {
+      console.log(`${label} took ${elapsed}ms to execute.`);
+    }
+
+    // Some portion of users will have timing information sent. Limit this further to
+    // only send a few labels per user.
+    if (_timingsPerLabel[label] <= MAX_TIMINGS_PER_LABEL) {
```

Comment:
When `_timingsPerLabel[label]` is `undefined` (the first time then), I think the result of this comparison is always `false`.

I think you need to set it to a temporary const before the condition, something like this:
```javascript
const sentTimings = _timingsPerLabel[label] || 0;
if (sentTimings < MAX_TIMINGS_PER_LABEL) {
  _timingsPerLabel[label] = sentTimings + 1;
  ...
}
```

Also note the operator `<` instead of `<=`. Otherwise I think it's sent (MAX_TIMINGS_PER_LABEL + 1) times.

Reply:
Thanks for catching this.